java/src/main/java/com/twisleton/gathering/services/GameService.java

```diff
@@ -34,6 +35,9 @@
     private final int worldMaxYCoordinate;
     private final Gson gson = new Gson();
 
+    private final double playerMoveDistance = 0.25;
+    private final double playerDiagonalMoveDistance = playerMoveDistance * Math.sqrt(2) / 2;
```

Comment:
why?

Reply:
I think I did math on this right. In the code I was playing with the move distance, so I changed it to 0.25, but for the sake simplicity let's just say that a cardinal direction movement is 1 unit. The player's movement should be the same regardless of which direction they are going in. Therefore, the total distance traveled by the player during a diagonal movement should still be 1 unit. There will be a vertical component to that and a horizontal component to that, so you can make a right triangle to show the change in position. The hypotenuse should be 1, so you derive the x and y deltas from that. Looking at this now, I guess it could simply be "`... = playerMoveDistance / Math.sqrt(2);`".

![182px-45-45-triangle svg](https://user-images.githubusercontent.com/5545241/91843552-ef3d3d80-ec23-11ea-9568-d970f42dea8b.png)

Reply:
Nicely worked out! Would've taken me ages 😂